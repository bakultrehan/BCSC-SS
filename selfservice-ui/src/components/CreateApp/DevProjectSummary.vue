/** * DevProjectSummary component */

<template>
  <v-card class="mx-auto outer-card">
    <v-toolbar flat class="bc-subtitle padding-0" dark>
      <v-btn icon @click="goBack()" :aria-label="$t('summaryPage.goBack')">
        <v-icon>mdi-arrow-left</v-icon>
      </v-btn>
      <v-toolbar-title>{{ $t('summaryPage.pagetitle') }}</v-toolbar-title>
      <div class="flex-grow-1"></div>

      <div class="flex-grow-1"></div>
    </v-toolbar>
    <v-container>
      <v-row>
        <v-col cols="8" flat>
          <v-card flat>
            <v-list-item-content
              class="text-left padding-0"
              v-html="$t('summaryPage.subTitle')"
            ></v-list-item-content>
          </v-card>
        </v-col>
        <v-col cols="4"><ProjectActions :id="id" /> </v-col>
        <ProjectSummary :id="id" />
      </v-row>
    </v-container>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import ProjectSummary from '@/components/CreateApp/ProjectSummary.vue';
import ProjectActions from '@/components/ProjectActions/ProjectActions.vue';

const SharedModule = namespace('SharedModule');
const ProjectInfoModule = namespace('ProjectInfoModule');

@Component({
  components: {
    ProjectSummary,
    ProjectActions,
  },
})
export default class DevProjectSummary extends Vue {
  @Prop({ default: 0 })
  public id!: number;

  @SharedModule.Action('redirectFromSummaryPage')
  public redirectFromSummaryPage!: any;
  @ProjectInfoModule.Getter('getSingleProjectInfo')
  public getSingleProjectInfo!: any;

  private projectId: number = this.id || 0;

  private mounted() {
    this.redirectFromSummaryPage(true);
  }

  private goBack() {
    this.redirectFromSummaryPage(false);
    this.$router.push(`/project/${this.projectId}/test-account`);
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.active-bg {
  background-color: $BCgovActiveBg;
}
.card-width {
  max-width: 110px;
}
.text-center {
  text-align: center !important;
}
</style>
