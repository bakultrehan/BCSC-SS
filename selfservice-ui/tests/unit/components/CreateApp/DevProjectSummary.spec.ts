import { shallowMount, mount } from '@vue/test-utils';
import DevProjectSummary from '@/components/CreateApp/DevProjectSummary.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
let vuetify: any;
vuetify = new Vuetify();

describe('DevProjectSummary.vue', () => {
  const store = new Vuex.Store({
    modules: {
      SharedModule: {
        namespaced: true,
        state: {},
        getters: {
          isRedirectFromSummaryPage: jest.fn()
        },
        actions: {
          redirectFromSummaryPage: jest.fn()
        }
      }
    }
  });

  const mountFunction = (options: any) => {
    return mount(DevProjectSummary, {
      store,
      vuetify,
      mocks: { $t: jest.fn(() => {}) }, // tslint:disable-line
      ...options
    });
  };

  it('renders props when passed with gettors', () => {
    const DevProjectSummaryPage = mountFunction({});

    expect(DevProjectSummaryPage.element).toMatchSnapshot();
  });

  // it('show modal on click', () => {
  //   const DevProjectSummary = mountFunction({});

  //   DevProjectSummary.setData({ isLoading: false });
  //   const button = DevProjectSummary.find('.submit-package');
  //   DevProjectSummary.vm.$on('action-btn:clicked', jest.fn());
  //   button.trigger('click');
  //   expect(DevProjectSummary.element).toMatchSnapshot();
  // });
});

// describe('DevProjectSummary.vue', () => {
//   it('renders props when passed', () => {
//     const store = new Vuex.Store({
//       modules: {
//         ProjectInfoModule: {
//           namespaced: true,
//           state: {},
//           getters: {
//             getSingleProjectInfo: jest.fn(),
//             errorStatus: jest.fn(),
//             getFinalProjectSubmissionStatus: jest.fn()
//           },
//           actions: {
//             loadSingleProjectInfo: jest.fn()
//           }
//         },
//         TechnicalReqModule: {
//           namespaced: true,
//           state: {},
//           getters: {
//             getTechnicalReq: jest.fn(),
//             errorStatus: jest.fn()
//           },
//           actions: {
//             loadTechnicalReqDetails: jest.fn()
//           }
//         },
//         PackageAndTestModule: {
//           namespaced: true,
//           state: {},
//           getters: {
//             successStatus: jest.fn(),
//             getPackageList: jest.fn()
//           },
//           actions: {
//             loadPackage: jest.fn()
//           }
//         },
//         SharedModule: {
//           namespaced: true,
//           state: {},
//           getters: {
//             isRedirectFromSummaryPage: jest.fn()
//           },
//           actions: {
//             redirectFromSummaryPage: jest.fn()
//           }
//         }
//       }
//     });
//     const DevProjectSummary = shallowMount(DevProjectSummary, {
//       vuetify,
//       store,
//       mocks: { $t: jest.fn(() => {}) } // tslint:disable-line
//     });

//     expect(DevProjectSummary.element).toMatchSnapshot();
//   });
// });
