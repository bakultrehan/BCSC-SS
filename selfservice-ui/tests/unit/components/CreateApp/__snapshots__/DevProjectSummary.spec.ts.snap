// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevProjectSummary.vue renders props when passed with gettors 1`] = `
<v-card-stub
  class="mx-auto outer-card"
  loaderheight="4"
  tag="div"
>
  <v-toolbar-stub
    class="bc-subtitle padding-0"
    dark="true"
    extensionheight="48"
    flat="true"
    src=""
    tag="header"
    tile="true"
  >
    <v-btn-stub
      activeclass=""
      icon="true"
      tag="button"
      type="button"
    >
      <v-icon-stub>
        mdi-arrow-left
      </v-icon-stub>
    </v-btn-stub>
     
    <v-toolbar-title-stub>
      
    </v-toolbar-title-stub>
     
    <div
      class="flex-grow-1"
    />
     
    <div
      class="flex-grow-1"
    />
  </v-toolbar-stub>
   
  <v-container-stub
    tag="div"
  >
    <v-row-stub
      tag="div"
    >
      <v-col-stub
        cols="8"
        flat=""
        tag="div"
      >
        <v-card-stub
          flat="true"
          loaderheight="4"
          tag="div"
        >
          <v-list-item-content-stub
            class="text-left padding-0"
          />
        </v-card-stub>
      </v-col-stub>
       
      <v-col-stub
        cols="4"
        tag="div"
      >
        <projectactions-stub
          id="0"
        />
      </v-col-stub>
       
      <projectsummary-stub
        id="0"
      />
    </v-row-stub>
  </v-container-stub>
</v-card-stub>
`;
